pkg/resource/grant/hook.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package grant
+
+import (
+	"context"
+	"fmt"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+)
+
+// updateNotSupported returns a terminal error because KMS grant
+// resource does not support updates.
+func (rm *resourceManager) updateNotSupported(ctx context.Context,
+	desired *resource,
+	latest *resource,
+	diffReporter *ackcompare.Delta,
+) (*resource, error) {
+	return nil, ackerr.NewTerminalError(
+		fmt.Errorf("grant resource does not support updates"),
+	)
+}
```

Comment:
Weird that we don't handle this in a general way as part of the `generator.yaml`?

Reply:
There is a "not implemented" error which is returned by default. But in this case I wanted to return a specific "not supported" terminal error. 
I agree that It can be turned into a generator config.